### ISSUE TITLE
conda forge bulds failing

### ISSUE THREAD

**Issue by bernt-matthias** [tcsh-org/tcsh#99]
the build of the tcsh conda forge recipe fails sice version `v6_24_11` with:

```
/home/conda/feedstock_root/build_artifacts/tcsh_1710376699475/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/12.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: tc.func.o: in function `auto_lock':
tc.func.c:(.text.auto_lock+0x8d): undefined reference to `crypt'
```

is this something that might be explained by a change in tcsh (maybe a new dependency)? Or might the error be on the conda forge side?

I would also like to ask if someone from the tcsh community wants to join in maintaining the conda forge recipe. I just created the recipe but have the feeling that I have to little understanding of tcsh to do it well enough. Effort seems minimal, it's just looking over automatically created PRs for the updates. 

Edit: here is a link: https://github.com/conda-forge/tcsh-feedstock/pull/22

**Comment by suominen** [comment]
I don't think that failure comes from any change in the actual `tcsh` code between 6.24.10 and 6.24.11. No new dependencies were added. (And that applies to 6.24.12, too.)

The `autoconf` stuff was also updated, so it is _possible_ that that introduced something. However, I have built `tcsh` on Debian (4 releases), Ubuntu (5 releases), and NetBSD (2 releases and -current) and they all succeed.

I'm inclined to think it is an issue with conda forge specifically, but I have no evidence. (I tried looking behind the link you provided, but gave up as I was unable to unearth actual build logs. Sorry...)

On `glibc` systems, `tcsh` is linked with the `crypt` library:

```
; ldd /bin/tcsh
        linux-vdso.so.1 (0x00007fff9ed26000)
        libtinfo.so.6 => /lib/x86_64-linux-gnu/libtinfo.so.6 (0x00007f396b54e000)
        libcrypt.so.1 => /lib/x86_64-linux-gnu/libcrypt.so.1 (0x00007f396b512000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f396b331000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f396b60c000)
```

Running `configure` should pick it up.

For reference, the full diff between the releases can be found here:

- https://github.com/tcsh-org/tcsh/compare/TCSH6_24_10...TCSH6_24_11

It is long due to the `autoconf`-related changes.
